tests/test.py

```diff
@@ -0,0 +1,139 @@
+        room =self.dojo.create_room("office", ["Bujumbura"])
+        self.assertEqual(room,"Room with name bujumbura already exists \n")
+
+    def test_create_wrong_room_type(self):
+        """
+        Tests adding wrong room type
+        should not increase the number of rooms
+
+        """
+        current_no_rooms = len(self.dojo.all_rooms)
+        room = self.dojo.create_room("social_hall", ["Stam"])
+        new_no_rooms = len(self.dojo.all_rooms)
+        self.assertEqual(room, "Wrong room type \n")
+        self.assertEqual(current_no_rooms, new_no_rooms)
+
+class TestAddingPersons(unittest.TestCase):
+    """Test cases for Adding Persons"""
+    def setUp(self):
+        self.dojo = Dojo()
+
+    def test_successfully_adds_fellow_to_system(self):
+        """
+        Should add fellow to the system
+        and increase the number of fellows
+
+        """
+        current_no_fellows = len(self.dojo.fellows)
+        self.dojo.add_person('Kipyegon','ken', 'Fellow')
+        person = Fellow('Maina','Wekesa')
+        self.assertIsInstance(person, Fellow)
+        new_no_fellows = len(self.dojo.fellows)
+        self.assertEqual(new_no_fellows-current_no_fellows, 1)
+
+    def test_successfully_adds_staff_to_system(self):
+        """ Should add staff to the system """
+        current_no_staff = len(self.dojo.staffs)
+        self.dojo.add_person('Shem','Ogube', 'Staff')
+        person = Staff('Ken','Kipyegon')
+        self.assertIsInstance(person, Staff)
+        new_no_staff = len(self.dojo.staffs)
+        self.assertEqual(new_no_staff-current_no_staff, 1)
+
+    def test_adding_person_with_invalid_category(self):
+        """Tests adding person who's category is not staff or fellow"""
+        new = self.dojo.add_person("Wekesa','Maina", "Bootcamper", "N")
+        self.assertEqual(new,"Wrong category. Can only be fellow or staff")
+
+class TestPrintingRoom(unittest.TestCase):
+    """ Test cases for printing Room and its occupants """
+    def setUp(self):
+        self.dojo = Dojo()
+
+    def test_print_room_that_does_not_exist(self):
+        """ Should not print a room that is not in the system """
+        room_print = self.dojo.print_room('Entebbe')
+        self.assertEqual(room_print, "Room name Entebbe doesn't exist")
+
+    def test_prints_room_occupants(self):
+        """ Should print all the occupants of a specified room """
+        self.dojo.create_room("office", ["Django"])
+        self.dojo.add_person('Dennis', 'Wachiuri', 'Fellow')
+        room=self.dojo.print_room('Django')
+        self.assertIn("Dennis Wachiuri", "room")
+
+    def test_prints_empty_for_unoccupied_room(self):
+        """ Prints null if specified room is empty """
+        self.dojo.create_room("office", ["Django"])
+        r_print = self.dojo.print_room('Django')
+        self.assertEqual(r_print, "Django has no occupants")
+
+class TestPrintAllocatedUnallocated(unittest.TestCase):
```

Comment:
Maybe consider creating a new method to test unallocated seperately as you have named the class 'TestPrintAllocatedUnallocated' but only defined one method testing allocations